modules/trusted-firmware-m/Kconfig.tfm

```diff
@@ -188,6 +188,7 @@ config TFM_BL2_NOT_SUPPORTED
 
 config TFM_IMAGE_VERSION_S
 	string "Version of the Secure Image"
+	default "$(APP_VERSION_TWEAK_STRING)" if "$(VERSION_MAJOR)" != "" && TFM_MCUBOOT_IMAGE_NUMBER = 1
```

Comment:
I am not sure about the uses of `TFM_MCUBOOT_IMAGE_NUMBER`. Why have this condition at all? Especially here, I imagine that we'd want the version also when `TFM_MCUBOOT_IMAGE_NUMBER = 2`.

Reply:
IMO, when split into two separate applications, the secure application shouldn't be versioned according to the version of the non-secure application. The default when `TFM_MCUBOOT_IMAGE_NUMBER = 1` is only set because `TFM_IMAGE_VERSION_S` is repurposed as the version of the combined application in that mode.

Reply:
It is not used at all when `TFM_MCUBOOT_IMAGE_NUMBER = 1`